openedx/core/djangoapps/schedules/tasks.py

```diff
@@ -309,8 +314,98 @@ def _upgrade_reminder_schedules_for_bin(site, target_day, bin_num, org_list, exc
         yield (user, first_schedule.enrollment.course.language, template_context)
 
 
+class CourseUpdate(ScheduleMessageType):
+    pass
+
+
+@task(ignore_result=True, routing_key=ROUTING_KEY)
+def course_update_schedule_bin(
+    site_id, target_day_str, day_offset, bin_num, org_list, exclude_orgs=False, override_recipient_email=None,
+):
+    target_day = deserialize(target_day_str)
+    msg_type = CourseUpdate()
+
+    for (user, language, context) in _course_update_schedules_for_bin(
+        Site.objects.get(id=site_id),
```

Comment:
How many iterations can this go up to?

Reply:
Currently, we're defaulting to 24 bins. But I was about to make the bin counts more dynamic (by estimating the number of schedules-per-day at runtime), in order to make the number of items per bin have a more definite cap.